apps/studio/data/api-keys/api-key-create-mutation.ts

```diff
@@ -0,0 +1,84 @@
+import { useMutation, UseMutationOptions, useQueryClient } from '@tanstack/react-query'
+import { toast } from 'sonner'
+
+import { handleError, post } from 'data/fetchers'
+import type { ResponseError } from 'types'
+import { apiKeysKeys } from './keys'
+
+export type APIKeyCreateVariables = {
+  projectRef?: string
+  description: string
+} & (
+  | {
+      type: 'publishable'
+    }
+  | {
+      type: 'secret'
+      // secret_jwt_template?: { // @mildtomato (Jonny) removed this field to reduce scope
+      //   role: string
+      // } | null
+    }
+)
+
+export async function createAPIKey(payload: APIKeyCreateVariables) {
+  if (!payload.projectRef) throw new Error('projectRef is required')
+
+  const { data, error } = await post('/v1/projects/{ref}/api-keys', {
+    params: {
+      path: { ref: payload.projectRef },
+      // WRONG QUERY PARAMETER
```

Comment:
just to check what this one is?

Reply:
ahh I think it was because it didn't make sense to reveal the API secret.
and the query param should be optional

Reply:
I'll just remove the comment